### PR TITLE
Add object lock configuration to augment table

### PR DESCRIPTION
**Pull request by dmytro-afanasiev**
```yaml
policies:
    - name: example
      resource: aws.s3
      filters:
        - type: lock-configuration
          key: Rule.DefaultRetention.Mode
          value: COMPLIANCE
```

### CODE REVIEW

**Comment by kapilt** [comment]
this should definitely be a separate filter, we're trying to slim augment table not add to it for s3, re not forcing api calls when they aren't explicitly needed for the policy.

**Comment by dmytro-afanasiev** [comment]
> this should definitely be a separate filter, we're trying to slim augment table not add to it for s3, re not forcing api calls when they aren't explicitly needed for the policy.

changed to value filter

File: c7n/resources/s3.py [hunk 1]
```diff
@@ -839,6 +839,65 @@ def get_std_format_args(self, bucket):
...
+            for f in as_completed(futures):
+                exc = f.exception()
+                if exc:
+                    self.log.error(
+                        "Exception getting bucket lock configuration \n %s" % (
+                            exc))
+        return super().process(resources, event)
+
+    def __call__(self, r):
+        if self.annotate:
```
Comment (kapilt) [inline c7n/resources/s3.py:896]: afaics given the class value for annotate = true, there's no need for the conditional here.

File: c7n/resources/s3.py [hunk 2]
```diff
@@ -839,6 +839,65 @@ def get_std_format_args(self, bucket):
...
+                    filters:
+                      - type: lock-configuration
+                        key: Rule.DefaultRetention.Mode
+                        value: COMPLIANCE
+
+    """
+    schema = type_schema('lock-configuration', rinherit=ValueFilter.schema)
+    permissions = ('s3:GetBucketObjectLockConfiguration',)
+    annotate = True
+    annotation_key = 'ObjectLockConfiguration'
```
Comment (kapilt) [inline c7n/resources/s3.py:865]: generally want a c7n: prefix on annotation keys to distinguish from native attributes.

**Review by kapilt — APPROVED** [review]
lgtm, thanks

**Pull request merged**